crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -99,15 +99,15 @@ where
     Pool: TransactionPool + 'static,
 {
     /// Installs a new filter and returns the new identifier.
-    async fn install_filter(&self, kind: FilterKind) -> RpcResult<SubscriptionId<'static>> {
+    async fn install_filter(&self, kind: FilterKind) -> RpcResult<FilterId> {
         let last_poll_block_number = self.client.chain_info().to_rpc_result()?.best_number;
         let id = self.id_provider.next_id();
         let mut filters = self.active_filters.inner.lock().await;
         filters.insert(
             id.clone(),
             ActiveFilter { last_poll_block_number, last_poll_timestamp: Instant::now(), kind },
         );
-        Ok(id)
+        Ok(id.into())
```

Comment:
while we're at it, let's change the key of the `active_filters`  to FilterId as well.